src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts

```diff
@@ -74,9 +74,9 @@ export class ServerTreeDragAndDrop implements IDragAndDrop {
 		const data = dragAndDropData.getData();
 		const element = data[0];
 		if (element.nodeTypeId === 'Column' || element.nodeTypeId === 'Table') {
-			const schema = element.metadata.schema;
-			const name = element.metadata.name;
-			originalEvent.dataTransfer.setData(DataTransfers.RESOURCES, JSON.stringify([`${element.nodeTypeId}:${element.id}?${schema ? schema + '.' + name : name}`]));
+			const escapedSchema = element.metadata.schema?.replace(/]/g, ']]');
+			const escapedName = element.metadata.name?.replace(/]/g, ']]');
+			originalEvent.dataTransfer.setData(DataTransfers.RESOURCES, JSON.stringify([`${element.nodeTypeId}:${element.id}?${escapedSchema ? escapedSchema + '].[' + escapedName : escapedName}`]));
```

Comment:
And if we're going to be adding the middle brackets here then I don't think we should be doing the outer brackets here : https://github.com/microsoft/azuredatastudio/pull/10702/files#diff-d38769f349f31d56f614f8337e3a8965R363

Is there any reason you couldn't just do the whole quoted name here?

Reply:
Shouldn't we have this escape logic in some common place so other parts of ads can access them?

Reply:
Potentially useful. Not really a huge deal but yeah having something that escapes and/or quotes identifiers might be useful enough to warrant making a helper method for.